synapse/lib/stormlib/imap.py

```diff
@@ -526,14 +526,22 @@ async def connect(self, host, port=imaplib.IMAP4_SSL_PORT, timeout=30, ssl=True,
 
         try:
             imap = await s_common.wait_for(coro, timeout)
-        except asyncio.TimeoutError:
+        except TimeoutError:
             raise s_exc.TimeOut(mesg='Timed out waiting for IMAP server hello.') from None
 
         async def fini():
```

Comment:
we should clean this up to use `onfini()` handlers rather than a `fini()` override.